src/main/java/org/openrewrite/java/migrate/lombok/LombokValueToRecord.java

```diff
@@ -99,23 +99,28 @@ public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, Ex
                 return cd;
             }
 
-            assert cd.getType() != null : "Class type must not be null";
+            assert cd.getType()!=null:"Class type must not be null";
+            Set<String> memberVariableNames = getMemberVariableNames(memberVariables);
+            if (implementsConflictingInterfaces(cd, memberVariableNames)) {
+                return cd;
+            }
+
             acc.putIfAbsent(
                     cd.getType().getFullyQualifiedName(),
-                    getMemberVariableNames(memberVariables));
+                    memberVariableNames);
 
             return cd;
         }
 
         private boolean isRelevantClass(J.ClassDeclaration classDeclaration) {
-            return classDeclaration.getType() != null
+            List<J.Annotation> allAnnotations = classDeclaration.getAllAnnotations();
+            return classDeclaration.getType()!=null
                     && !J.ClassDeclaration.Kind.Type.Record.equals(classDeclaration.getKind())
-                    && classDeclaration.getAllAnnotations().stream()
-                    .allMatch(ann -> LOMBOK_VALUE_MATCHER.matches(ann) && (ann.getArguments() == null || ann.getArguments().isEmpty()))
+                    && !allAnnotations.isEmpty()
```

Comment:
This is necessary because `allMatch` will return `true` on an empty set. The recipe would then convert an empty interface to a record. Not sure why that did not cause issue before.